### PR TITLE
hide other password args

### PR DESCRIPTION
**Pull request by xhebox**
### What problem does this PR solve? 

As title.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 ```
[xhe@M1PRO-2 tiup]$ ./bin/tiup tidb-lightning:v6.5.3 --tidb-password tidb
Starting component tidb-lightning: /Users/xhe/.tiup/components/tidb-lightning/v6.5.3/tidb-lightning --tidb-password ******
Verbose debug logs will be written to /var/folders/4q/x954lkc95gl34qjnx2hbcr640000gn/T/lightning.log.2024-07-09T14.51.18+0800
```
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
Redact `--tidb-password` arguments for tidb-lightning
```

### CODE REVIEW

File: pkg/environment/history.go [hunk 1]
```diff
@@ -245,17 +245,29 @@ func getLatestHistoryFile(dir string) (item historyItem) {
 
 // HidePassword replace password with ******
 func HidePassword(args []string) []string {
-	var record []string
+	redactArgs := []string{
+		"-p",
+		"--tidb-password",
```
Comment (bb7133) [inline pkg/environment/history.go:250]: Do we have any other possible password arguments?

**Review by bb7133 — APPROVED** [review]
LGTM

File: pkg/environment/history.go (hunk 1 above)
Comment (xhebox) [reply pkg/environment/history.go:254]: Added another for dumpling, and comments, PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**